### DISCUSSION TITLE
AppRouter v14.1.0  Stuck in _rsc fetch loop HELL!

### DISCUSSION THREAD

**Discussion by FadiObaji** [vercel/next.js#61613] [Help]
### Summary

Hello there..

I've been working on this website for a month now, and two days ago while testing I discovered this weird thing that only happens with ONE route in the website.

![Screenshot 2024-02-03 155951](https://github.com/vercel/next.js/assets/24692864/3081d816-3cab-4002-a4f3-2762d0eac9e4)

**This is the thing**
There is a component called 'DistrictCard.tsx' a small component that I use to display the city districts whenever I want to display them (usng map.()) and it has a **Link** to take you to this route I'm talking about.

The route works for the first time just fine, meaning: clicking on the card will take me to the page nicely. 
However, when navigating to any other page that has those same cards displayed somewhere, and then clicking on a DistrictCard.tsx again, the above Hell happens and the page never opens! Visually the website looks normal and you just keep clicking on this link and it never opens anything. But if you click on any other 'Link's they work just fine.

I tried everything from changing the names to changing the routes to removing the middleware to removing the redirects in the next.config.... Nothing solved this!

Please any help is appreciated here!

What am I missing?

**Comment by cherealnice** [comment]
Did you have any success in figuring this out? Having the same issue

**Reply by cherealnice** [reply]
Are you by chance using generateMetadata on your pages?

**Reply by FadiObaji** [reply]
Yeah, I'm using generateMetadata, but not in the page that is causing this error. Why? Do you know how to solve this?


> Did you have any success in figuring this out? Having the same issue

I ended up removing the components from the homepage and only use it in one page.
It's a pretty weird behavior!

**Reply by cherealnice** [reply]
I think the issue I'm having is related to generateMetadata but it's hard to tell. Removing it seems to solve the problem, but my current thought is that it's happening due to the generateMetadata function taking some time to load its data. Is your component async?

**Reply by FadiObaji** [reply]
The main component is async that has the client component imported in it. But I don't think this is the problem.
